src/odemis/acq/test/stream_test.py

```diff
@@ -2430,16 +2289,164 @@ def test_acq_arpol_leech(self):
+        ars = stream.ARSettingsStream("test ar with analyzer", self.ccd, self.ccd.data, self.ebeam, self.analyzer)
+
+        sas = stream.SEMARMDStream("test sem-ar", [sems, ars])
+
+        ars.polarization.value = "vertical"
+        ars.roi.value = (0, 0.2, 0.3, 0.6)
+        dc = leech.AnchorDriftCorrector(self.ebeam, self.sed)
+        dc.period.value = 1  # s
+        dc.roi.value = (0.525, 0.525, 0.6, 0.6)
+        dc.dwellTime.value = 1e-06
+        sems.leeches.append(dc)
+
+        # Short acquisition (< 0.1s)
+        self.ccd.exposureTime.value = 0.03  # s
+        ars.repetition.value = (10, 5)
+        exp_pos, exp_pxs, exp_res = self._roiToPhys(ars)
+        phys_roi = (exp_pos[0] - (exp_pxs[0] * exp_res[0] / 2),
+                    exp_pos[1] - (exp_pxs[1] * exp_res[1] / 2),
+                    exp_pos[0] + (exp_pxs[0] * exp_res[0] / 2),
+                    exp_pos[1] + (exp_pxs[1] * exp_res[1] / 2),
+                    )
+
+        num_ar = numpy.prod(ars.repetition.value)
+
+        # Start acquisition
+        # estimated acquisition time should be accurate with less than 50% margin + 1 extra second
+        timeout = 1 + 1.5 * sas.estimateAcquisitionTime()
+        start = time.time()
+        f = sas.acquire()
+
+        # sas.raw: array containing as first entry the sem scan image for the scanning positions,
+        # rest are ar images
+        # data: array should contain same images as sas.raw
+
+        # wait until it's over
+        data = f.result(timeout)
+        dur = time.time() - start
+        logging.debug("Acquisition took %g s", dur)
+        self.assertTrue(f.done())
+
+        # check if number of images in the received data (sem image + ar images) is the same as
+        # number of images stored in raw
+        self.assertEqual(len(data), len(sas.raw))
+
+        # check that sem data array has same shape as expected for the scanning positions of ebeam
+        sem_da = sas.raw[0]  # sem data array for scanning positions
+        self.assertEqual(sem_da.shape, exp_res[::-1])
+
+        # check that number of angle resolved images is same as the total number of ebeam positions
+        # include if multiple polarization images are required per ebeam position
+        ar_das = sas.raw[1:]  # angle resolved data arrays
+        if sas._polarization.value == POL_6POS:
+            self.assertEqual(len(ar_das), num_ar*6)
+        else:
+            self.assertEqual(len(ar_das), num_ar)
+
+        # TODO
+        # check if metadata is correctly stored
+        for d in ar_das:
+            md = d.metadata
+            # check if model.MD_POS is in metadata
+            self.assertIn(model.MD_POS, md)
+            self.assertIn(model.MD_AR_POLE, md)
+            pos = md[model.MD_POS]
+            self.assertTrue(phys_roi[0] <= pos[0] <= phys_roi[2] and
+                            phys_roi[1] <= pos[1] <= phys_roi[3])
+
+    def test_acq_arpol_leech(self):
```

Comment:
The function is defined twice! (One of them is discarded when Python loads it) => drop the duplicate function.